aten/src/ATen/native/MathBitsFallback.cpp

```diff
@@ -0,0 +1,119 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <torch/library.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/native/UnaryOps.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+
+struct MathOpFallback {
+  MathOpFallback(DispatchKey key_, string op_name_) : key(key_), op_name(op_name_) {}
+  virtual bool is_bit_set(c10::TensorImpl&) = 0;
+  virtual void set_bit(Tensor&, bool) = 0;
```

Comment:
These (all the `Tensor&` references) also don't have to be mutable references, but for a different reason: const-correctness on Tensor is a lie and `const Tensor&` will always have the same set of methods available.

Reply:
https://github.com/zdevito/ATen/issues/27#issuecomment-330717839